meson.build

```diff
@@ -0,0 +1,170 @@
+    '/D_CRT_SECURE_CPP_OVERLOAD_STANDARD_NAMES_COUNT=1',
+    '/D_CRT_NONSTDC_NO_WARNINGS=1',
+    '/D_CRT_SECURE_NO_WARNINGS=1',
+    language : ['c'],
+  )
+endif
+
+# TODO: solaris extensions
+
+conf = configuration_data()
+
+foreach h : ['assert.h', 'inttypes.h', 'io.h', 'malloc.h', 'memory.h',
+             'setjmp.h', 'signal.h', 'stdarg.h', 'stddef.h', 'stdint.h',
+             'stdio.h', 'stdlib.h', 'string.h', 'strings.h', 'sys/stat.h',
+             'sys/types.h', 'time.h', 'unistd.h']
+  if cc.check_header(h)
+    conf.set('HAVE_@0@'.format(h.underscorify().to_upper()), 1)
+  endif
+endforeach
+
+if conf.get('HAVE_TIME_H', 0) == 1
+  if cc.has_member('struct timespec', 'tv_sec', prefix : '#include <time.h>')
+    conf.set('HAVE_STRUCT_TIMESPEC', 1)
+  endif
+endif
+
+foreach f : ['calloc', 'exit', 'fprintf', 'free', 'longjmp', 'siglongjmp',
+             'malloc', 'memcpy', 'memset', 'printf', 'setjmp', 'signal',
+             'strsignal', 'strcmp', 'clock_gettime']
+  if cc.has_function(f)
+    conf.set('HAVE_@0@'.format(f.underscorify().to_upper()), 1)
+  endif
+endforeach
+
+if host_machine.system() == 'windows'
+  foreach f : ['_vsnprintf_s', '_vsnprtinf', '_snprintf_s', '_snprintf']
+    if cc.has_function(f)
+      conf.set('HAVE_@0@'.format(f.underscorify().to_upper()), 1)
+    endif
+  endforeach
+  foreach f : ['snprintf', 'vsnprintf']
+    if cc.has_header_symbol('stdio.h', f)
+      conf.set('HAVE_@0@'.format(f.underscorify().to_upper()), 1)
+    endif
+  endforeach
+else
+  foreach f : ['snprintf', 'vsnprintf']
+    if cc.has_function(f)
+      conf.set('HAVE_@0@'.format(f.underscorify().to_upper()), 1)
+    endif
+  endforeach
+endif
+
+if host_machine.system() == 'windows'
+  if cc.compiles('''
+      __declspec(thread) int tls;
+
+      int main(void) {
+        return 0;
+      }''',
+      name : 'Thread Local Storage')
+    conf.set('HAVE_MSVC_THREAD_LOCAL_STORAGE', 1)
+  endif
+else
+  if cc.compiles('''
+      __thread int tls;
+
+      int main(void) {
+        return 0;
+      }''',
+      name : 'Thread Local Storage')
+    conf.set('HAVE_GCC_THREAD_LOCAL_STORAGE', 1)
+  endif
+endif
+
+dep_rt = cc.find_library('rt', required : false)
+
+if (conf.get('HAVE_TIME_H', 0) == 1 and
+    conf.get('HAVE_STRUCT_TIMESPEC', 0) == 1 and
+    conf.get('HAVE_CLOCK_GETTIME', 0) == 1)
+  if cc.has_header_symbol('time.h', 'CLOCK_REALTIME')
+    conf.set('HAVE_CLOCK_REALTIME', 1)
+  endif
+endif  
+
+conf.set('WORDS_SIZEOF_VOID_P', cc.sizeof('void *'))
+if host_machine.endian() == 'big'
+  conf.set('WORDS_BIGENDIAN', 1)
+endif
+
+# TODO: pkg-config
+# TODO: cmake-config
+
+inc_include = include_directories('include')
+
+subdir('private')  # someplace safe to put the generated config.h file
+subdir('src')
+
+# TODO: doc, include, tests, example
+# Since we're using this as a wrap, and it's a unit test framework we're not
+# going to install it.
+
+dep_cmocka = declare_dependency(
```

Comment:
It would be nice to name this variable `cmocka_dep` according to the [meson style](https://mesonbuild.com/Dependencies.html#dependencies).